version: "3.8"

services:
  rabbitmq:
    image: rabbitmq:3.12-management
    ports:
      - "5672:5672"
      - "15672:15672"
    environment:
      RABBITMQ_DEFAULT_USER: admin
      RABBITMQ_DEFAULT_PASS: admin
    healthcheck:
      test: ["CMD", "rabbitmqctl", "status"]
      interval: 10s
      timeout: 5s
      retries: 5
    networks:
      - app-network

  order_service:
    build:
      context: .
      dockerfile: ./apps/order_service/Dockerfile
    env_file:
      - ./apps/order_service/.env
    depends_on:
      rabbitmq:
        condition: service_healthy
    ports:
      - "4002:4002"
    networks:
      - app-network

  products_service:
    build:
      context: .
      dockerfile: ./apps/products_service/Dockerfile
    env_file:
      - ./apps/products_service/.env
    depends_on:
      rabbitmq:
        condition: service_healthy
    ports:
      - "4003:4003"
    networks:
      - app-network

  auth_service:
    build:
      context: .
      dockerfile: ./apps/auth_service/Dockerfile
    env_file:
      - ./apps/auth_service/.env
    depends_on:
      rabbitmq:
        condition: service_healthy
    ports:
      - "4001:4001"
    networks:
      - app-network

networks:
  app-network:
    driver: bridge
